nrich-search/README.md

```diff
@@ -150,76 +176,222 @@ public class CarSearchResult {
-but this cane be overridden either on type level or on property level by using `searchOperatorOverrideList`. Queries on plural associations are done using exists query (to avoid duplicate results) but
-that can be overridden by property `pluralAssociationRestrictionType`. Additional restrictions (not dependent on data in search request, for example security restrictions) can be specified
-using `additionalRestrictionResolverList`. When searching also by entity that doesn't have a direct association to root entity `subqueryConfigurationList` property needs to be defined.
-`SearchConfigration` also supports matching any value (or operator is used when creating query) by setting `anyMatch` parameter to `true` or matching all values (default behaviour) and operator is
-used for query.
-
-`subqueryConfigurationList` is useful when we want to search entities without direct association i.e. if we would like to search User based on a Role and User has no direct association to role instead
-we mapped that connection by using UserRole entity.
+### Using NaturalIdSearchExecturor
+
+[`NaturalIdSearchExecturor`][natural-id-search-executor-url] only works with Hibernate JPA implementation and enables users to search entities by properties annotated with `@NaturalId` annotation.
+It is useful when entity has, beside its primary key, a natural key (akin to `registrationNumber` in entity `Car`).
+
+### Combining search executors
+
+Users are able to use all types of search executors together on one JPA repository. This is achieved by extending [`SearchExecutor`][search-executor-url],
+[`StringSearchExecutor`][string-search-executor-url] and [`NaturalIdSearchExecturor`][natural-id-search-executor-url] in user defined JPA repository.
+
+An example of such a repository would be:
+
+```java
+public interface CarSearchDemoRepository extends JpaRepository<Car, Long>, SearchExecutor<Car>, StringSearchExecutor<Car>, NaturalIdSearchExecutor<Car> {
 
+}
 ```
 
-    SubqueryConfiguration subqueryConfiguration = SubqueryConfiguration.builder()
-                .rootEntity(UserRole.class)
-                .propertyPrefix("userRole")
-                .joinBy(new SearchPropertyJoin("id", "user.id")).build();
+## SearchConfiguration
+
+[`SearchConfiguration`][search-configuration-url] has a lot of options that let users customize query creation and result resolution.
+
+##### Defining result class
+
+If a regular class needs to be returned instead of projection,`.resultClass(SearchCarResult.class)` can be omitted and return class will be resolved from repository type parameter.
+Instead of a result class users can also manually define a list of projections using `projectionList` property.
+
+#### Defining table joins
+
+Users can define joins (fetches) (so no data is fetched in one select) by using `joinList`.
+Supported joins are **inner joins** and **left outer joins**.
```

Comment:
Maybe just a little change to `Supported joins are inner join and left outer join`

Reply:
Done.